src/auxiliary/auxiliary.jl

```diff
@@ -345,4 +345,16 @@ function register_error_hints()
 
     return nothing
 end
+
+function retrieve(file_path, src_url)
```

Comment:
This function is not specific to t8code, right? Would it make sense to use `Trixi.retrieve` in all examples, where `isfile(file_path) || download(src_url, file_path)` is used now? Additionally, I think this needs some docs when we use this in the examples.